schema/m3-room.schema.json

```diff
@@ -0,0 +1,1922 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "id": "https://raw.githubusercontent.com/vg-json-data/sm-json-data/master/schema/m3-room.schema.json",
+  "definitions": {
+    "runway": {
+      "$id": "#/definitions/runway",
+      "type": "object",
+      "required": ["length"],
+      "additionalProperties": false,
+      "properties": {
+        "length": {
+          "$id": "#/definitions/runway/properties/length",
+          "type": "number",
+          "title": "Length",
+          "minimum": 0,
+          "maximum": 45,
+          "description": "Horizontal length of the runway in tiles."
+        },
+        "openEnd": {
+          "$id": "#/definitions/runway/properties/openEnd",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 2,
+          "title": "Number of Open Ends",
+          "description": "The number of open ends in the runway. An open end is a runway edge that ends in a dropoff rather than a wall. Each open end adds a bit of extra room.",
+          "default": 1
+        },
+        "gentleUpTiles": {
+          "$id": "#/definitions/runway/properties/gentleUpTiles",
+          "type": "integer",
+          "title": "Gentle Up Tiles",
+          "description": "Number of tiles in the runway that gently slope (half a vertical tile) upwards"
+        },
+        "gentleDownTiles": {
+          "$id": "#/definitions/runway/properties/gentleDownTiles",
+          "type": "integer",
+          "title": "Gentle Down Tiles",
+          "description": "Number of tiles in the runway that gently slope (half a vertical tile) downwards"
+        },
+        "steepUpTiles": {
+          "$id": "#/definitions/runway/properties/steepUpTiles",
+          "type": "integer",
+          "title": "Steep Up Tiles",
+          "description": "Number of tiles in the runway that steeply slope (one vertical tile) upwards"
+        },
+        "steepDownTiles": {
+          "$id": "#/definitions/runway/properties/steepDownTiles",
+          "type": "integer",
+          "title": "Steep Down Tiles",
+          "description": "Number of tiles in the runway that steeply slope (one vertical tile) downwards"
+        }
+      }
+    },
+    "strat": {
+      "$id": "#/definitions/strat",
+      "type": "object",
+      "required": [
+        "name",
```

Comment:
Had to revert this: we can't make "from" and "to" required in the schema yet, since there are strats inside runways (etc.) that don't have them. After the cross-room strat migration is complete, we could add this to make those fields required.

In the meantime, I added checks in the keywords test now to give a nice error if "from" or "to" is missing for strats in the room "strats" array (previously the test would have just crashed if one of these were missing).